core/cpu.c

```diff
@@ -502,7 +502,15 @@ uint8 is_vmcs_loaded(struct vcpu_t *vcpu)
 
 int debug_vmcs_count = 0;
 
-void restore_host_cr4_vmxe(struct per_cpu_data *cpu_data);
+void check_vmxon_coherency(struct per_cpu_data *cpu_data)
```

Comment:
Make this function `static`? Actually, a better idea is to merge this function into `vmxroot_leave()` (after modifying `vmxroot_leave()` as I propose below).